src/Libraries/RevitNodes/Elements/FilledRegion.cs

```diff
@@ -0,0 +1,174 @@
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Reference to the Element
+        /// </summary>
+        [SupressImportIntoVM]
+        public override Autodesk.Revit.DB.Element InternalElement
+        {
+            get { return InternalRevitElement; }
+        }
+
+        #endregion
+
+        #region Private mutators
+
+        /// <summary>
+        /// Set the internal Element, ElementId, and UniqueId
+        /// </summary>
+        private void InternalSetElement(Autodesk.Revit.DB.FilledRegion element)
+        {
+            InternalRevitElement = element;
+            InternalElementId = element.Id;
+            InternalUniqueId = element.UniqueId;
+        }
+
+        #endregion
+
+        #region Private constructors
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="FilledRegion"></param>
+        private FilledRegion(Autodesk.Revit.DB.FilledRegion FilledRegion)
+        {
+            SafeInit(() => InitElement(FilledRegion));
+        }
+
+        /// <summary>
+        /// Create Filled Region by Curves
+        /// </summary>
+        /// <param name="view"></param>
+        /// <param name="typeId"></param>
+        /// <param name="boundary"></param>
+        private FilledRegion(View view, ElementId typeId, List<CurveLoop> boundary)
+        {
+            SafeInit(() => Init(view, typeId, boundary));
+        }
+
+        #endregion
+
+        #region Helpers for private constructors
+
+        /// <summary>
+        /// Init element
+        /// </summary>
+        /// <param name="element"></param>
+        private void InitElement(Autodesk.Revit.DB.FilledRegion element)
+        {
+            InternalSetElement(element);
+        }
+
+        /// <summary>
+        /// Init Filled Region by Curves
+        /// </summary>
+        /// <param name="view"></param>
+        /// <param name="typeId"></param>
+        /// <param name="boundary"></param>
+        private void Init(View view, ElementId typeId, IEnumerable<CurveLoop> boundary)
+        {
+            // Get document and start transaction
+            Autodesk.Revit.DB.Document document = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+
+            // Get existing element
+            var region = ElementBinder.GetElementFromTrace<Autodesk.Revit.DB.FilledRegion>(document);
+
+
+            if (region == null)
+            { 
+                // Create new region
+                region = Autodesk.Revit.DB.FilledRegion.Create(document, typeId, view.Id, boundary.ToList()); 
+            }
+            else
+            {
+                // Delete and replace region because there are no properties to update the filled region
+                document.Delete(region.Id);
+                region = Autodesk.Revit.DB.FilledRegion.Create(document, typeId, view.Id, boundary.ToList());
```

Comment:
I think you'll need to add some logic for comparing the inputs and bailing if they are the same to avoid infinite loops when in run auto mode. There should be some curve comparison logic in the curve utilities classes we already have.